tests/unit/test_conf.py

```diff
@@ -225,3 +226,32 @@ def test_max_key_fee_from_args(self):
         args = parser.parse_args(['--max-key-fee', '1.0', 'BTC'])
         c = Config.create_from_arguments(args)
         self.assertEqual(c.max_key_fee, {'amount': 1.0, 'currency': 'BTC'})
+
+    def test_one_of_string(self):
+        with self.assertRaises(ValueError):
+            no_vaid_values = OneOfString([], "no valid values", None)
+
+        with self.assertRaises(ValueError):
+            default_none = OneOfString(["a"], "invalid default", None)
+        with self.assertRaises(ValueError):
+            invalid_default = OneOfString(["a"], "invalid default", "b")
+
+        valid_default = OneOfString(["a"], "valid default", "a")
+
+        self.assertEqual("hello", OneOfString(["hello"], "valid default", "hello").default)
```

Comment:
^ would be better to do: `self.assertEqual("hello". TestConfig().one_of_string)` since that's testing how it's actually used everywhere